### PR TITLE
HubSpot to JSON - Companies

### PR DESCRIPTION
**Pull request by charlotta-frends**
#114

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters.
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots.
- [x] Template parameters checked
  - [x] Parameter descriptions are good and end with dots.
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output
- [x] End "balls" like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Tempalte files are going to the correct folder

## Summary by CodeRabbit

- **New Features**
  - Introduced "HubSpot to JSON - Companies" template for retrieving company data from HubSpot and saving it locally as a JSON file.
  - Added comprehensive documentation covering prerequisites, implementation details, usage notes, and error handling.
  - Released metadata and process templates for version 1.0.0.

### CODE REVIEW

File: Templates/CRMs/HubSpot/HubSpot to JSON - Companies/long-description.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+This template retrieves company data from HubSpot, and downloads it to a local fileshare as a single JSON file.
+
```
Comment (FrendsSarlinS) [inline Templates/CRMs/HubSpot/HubSpot to JSON - Companies/long-description.md:2]: Remove unnecessary comma here.

File: Templates/CRMs/HubSpot/HubSpot to JSON - Companies/long-description.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+![Template](assets/HubSpot_to_JSON_-_Companies.svg)
+
+# Prerequisites
+
+This template assumes that the following prerequisites are in place:
+
+- Using the HubSpot API requires an API key.
```
Comment (FrendsSarlinS) [inline Templates/CRMs/HubSpot/HubSpot to JSON - Companies/long-description.md:9]: The "Using the HubSpot API requires an API key" is not a prerequisite per se, rather a statement. Something like "HubSpot API key is accessible" is more like a prerequisite in this context.

**Comment by charlotta-frends** [comment]
@FrendsSarlinS Made these changes as well!

**Review by FrendsSarlinS — APPROVED** [review]
Approved!

File: Templates/CRMs/HubSpot/HubSpot to JSON - Companies/metadata.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "HubSpot to JSON - Companies",
+  "version": "1.0.0",
+  "tags": [],
+  "additionalSystemsUsed": []
```
Comment (jefim) [inline Templates/CRMs/HubSpot/HubSpot to JSON - Companies/metadata.json:5]: Mention HubSpot as we are using a HTTP task

**Review by jefim — CHANGES_REQUESTED** [review]

**Comment by charlotta-frends** [comment]
- Changed the template logic to accommodate the API's result pagination (the API calls return only 100 companies at maximum before the results are split into pages)
- Updated long description accordingly
- Included HubSpot as an additional system in metadata as requested

**Review by jefim — APPROVED** [review]

**Pull request merged**